release/RELEASE_PROCESS.md

```diff
@@ -2,48 +2,162 @@
+
+   You can copy the checklist above into the PR description in the next step.
+
+5. Submit your PR to the `release-*` branch that was created earlier and wait for
+   it to merge.
+
+6. Update the following files in the `main` branch with the same versions as in the
+   steps above, submit them in a PR to `main`, and wait for that PR to be merged:
+
+   - [ ] `docs/component-versions.md`
+   - [ ] `docs/quickstart.md`
+   - [ ] `docs/install/install-script.md`
+   - [ ] `scripts/setup_user_namespaces.sh`
+
+7. Generate the release manifests on the `release-*` branch:
+
+   ```Shell
+   kustomize build config/default > modelmesh.yaml
+   kustomize build config/runtimes --load-restrictor LoadRestrictionsNone > modelmesh-runtimes.yaml
+   cp config/dependencies/quickstart.yaml modelmesh-quickstart-dependencies.yaml
+   ```
+
+   If you see `Error: unknown flag: --load-restrictor` upgrade your `kustomize` version to 4.x.
+
+8. Generate config archive on the `release-*` branch. The scriptlet below automatically
+   determines the release version and chooses the version of the `tar` command for
+   either Linux or macOS. Verify the correct release `VERSION` was found.
+
+   ```Shell
+   VERSION=$( grep -o -E "newTag: .*$" config/manager/kustomization.yaml | sed 's/newTag: //' )
+   TAR_FILE="config-${VERSION}.tar.gz"
+
+   echo "Release: ${VERSION}"
+
+   if $(tar --version | grep -q 'bsd'); then
+     tar -zcvf ${TAR_FILE} -s /config/config-${VERSION}/ config/;
+   else
+     tar -zcvf ${TAR_FILE} config/ --transform s/config/config-${VERSION}/;
+   fi
+   ```
+
+9. Create a new tag on the `release-*` branch and push it to GitHub using the commands
+   below, or, create a new tag in the next step using the GitHub UI. The new
+   `kserve/modelmesh-controller` image will be published via GitHub Actions.
+
+   ```Shell
+   git tag $VERSION
+   git push upstream $VERSION
+
+   echo https://github.com/kserve/modelmesh-serving/releases/new?tag=${VERSION}
+   ```
+
+10. Create the new release in the GitHub UI from the `release-*` branch (or from the
+    tag created in the previous step). Enter the release tag value (i.e. `v0.10.0`) in
+    the "Release title" field and upload the generated install manifests ("Release assets")
+    in the "Atach binaries ..." section. Click the "Generate release notes" button which
+    will generate the release description.
+
+    **Note**, if you generated a pre-release (i.e. `v0.10.0-rc0`) then copy the release
+    notes from that and remove them from the pre-release description and revise accordingly.
+
+    https://github.com/kserve/modelmesh-serving/releases/new
+
+11. Compare the release and release artifacts to those of previous releases to make
+    sure nothing was missed. Verify that the newly released version of the
+    [`modelmesh-controller`](https://hub.docker.com/r/kserve/modelmesh-controller/tags)
+    was pushed to DockerHub.
```

Comment:
Should we mention anything about testing / checking fvt results?

Reply:
yup -- my running list:

- FVT, test -- verify the "check" results on last pushed tag
- update KServe ... website, charts (more details to be gathered)
- work with Dan on a relase blog